pkg/skaffold/schema/latest/v2/config.go

```diff
@@ -75,7 +79,425 @@ type Pipeline struct {
 	Deploy DeployConfig `yaml:"deploy,omitempty"`
 
 	// PortForward describes user defined resources to port-forward.
-	PortForward []*latestV1.PortForwardResource `yaml:"portForward,omitempty"`
+	PortForward []*PortForwardResource `yaml:"portForward,omitempty"`
+}
+
+// GitInfo contains information on the origin of skaffold configurations cloned from a git repository.
+type GitInfo struct {
```

Comment:
how will these be kept in sync with everyone making changes to the `v1.SkaffoldConfig` structs? We iterate very frequently on that config, and this could cause integration issues when we finally switch over from v2 to v3 config versions.

Reply:
Good point. Since render v2 is focused only on rend and deploy, I think we can update this field right before releasing v2 /v3schema. The reason to decouple v1 and v2 is to unblock the development of [v2 featured branch](https://github.com/GoogleContainerTools/skaffold/tree/v2), which only has v2 pipelines.

Reply:
@yuwenma to clarify, are you proposing that we manually update all duplicated fields between the old and new schemas before we release the new schema? this worries me a little bit as there are no guardrails in place preventing us from forgetting to do this.

Reply:
More context can be found in this [doc](https://docs.google.com/document/d/11UMjdKzfwVa6Y9I5lasR7x68PPixj_uIaNlxdIu7C2w/edit?resourcekey=0-F9iGb_D5vmd586bkSr-Hyw#), basically the render v2 will be developed in a featured branch "v2" and cherry-picking/merging-back to master. The featured branch v2 does not depends on v1 schemas (only v2 runnerContext and v3alpha schemas are used) and thus, we are doing the best effort to cherry-pick PRs into master. Agree decoupling v1 and v2 schema may add some additional work, but without this change, it will be even harder (if not impossible) to merge v2 back to master. 

The problem you mentioned are more depending on how skaffold plan to support version skews, which are orthogonal to the render v2 implementation and may deserve an individual design. @tejal29 @briandealwis Thoughts?

Reply:
all changes to build, Test and other config sections except (Deploy and render) will be carried over to v3alpha1 config. 
The v3alpha config has references to latestv1 config
```
// Pipeline describes a Skaffold pipeline.
type Pipeline struct {
	// Build describes how images are built.
	Build latestV1.BuildConfig `yaml:"build,omitempty"`

	// Test describes how images are tested.
	Test []*latestV1.TestCase `yaml:"test,omitempty"`

	// Render describes how the original manifests are hydrated, validated and transformed.
	Render RenderConfig `yaml:"manifests,omitempty"`

	// Deploy describes how the manifests are deployed.
	Deploy DeployConfig `yaml:"deploy,omitempty"`

	// PortForward describes user defined resources to port-forward.
	PortForward []*latestV1.PortForwardResource `yaml:"portForward,omitempty"`
}
```
Only Deploy and render config is changed.